cclib/method/ddec.py

```diff
@@ -631,67 +685,258 @@ def _calculate_w_and_u(self):
+        self._g = []
+
+        for atomi in range(self.data.natom):
+            # G_A -- equation S102 in doi: 10.1039/c6ra04656h
+            self._g.append(numpy.zeros_like(self.proatom_density[atomi], dtype=float))
+            self._g[atomi] = self.rho_wavg[atomi] + bigphiAII[atomi] * numpy.sqrt(
+                self.rho_wavg[atomi]
+            )
+            # Exponential constraint (as expressed in equation S105)
+            eta = (1 - (self.tau[atomi]) ** 2) * 1.75 * convertor(1, "Angstrom", "bohr")
+            exp_applied = self._g[atomi][:-1] * numpy.exp(
+                -1 * eta[1:] * numpy.diff(self.radial_grid_r[atomi])
+            )
+            for radiusi in range(1, len(self._g[atomi])):
+                self._g[atomi][radiusi] = min(self._g[atomi][radiusi], exp_applied[radiusi - 1])
+            # phi_A^II -- Equation S106 in doi: 10.1039/c6ra04656h
+            phiAII[atomi] = self._integrate_from_radial(
+                [self._g[atomi] - self.rho_wavg[atomi]], [atomi]
+            )
+
+            self._candidates_bigPhi.append([bigphiAII[atomi]])
+            self._candidates_phi.append([phiAII[atomi]])
+
+            phiconverged = True  # when convergence is reached, this is modified to False.
+
+            while phiAII[atomi] <= 0:
+                # Iterative algorithm until convergence
+                # Refer to Figure S3 in doi: 10.1039/c6ra04656h
+                temp = self._integrate_from_radial([numpy.sqrt(self.rho_wavg[atomi])], [atomi])
+                bigphiAII[atomi] = 2 * bigphiAII[atomi] - phiAII[atomi] / temp
+
+                # When Phi is updated, related quantities are updated as well
+                # Refer to S102 in doi: 10.1039/c6ra04656h
+                phiAII[atomi], self._g[atomi] = self._phiaii(
+                    self.rho_wavg[atomi], eta, bigphiAII[atomi], atomi
+                )
+
+                self._candidates_phi[atomi].append(phiAII[atomi])
+                self._candidates_bigPhi[atomi].append(bigphiAII[atomi])
+
+            # lowerbigPhi is largest negative Phi.
+            # upperbigPhi is smallest positive Phi.
+            if phiconverged:
+                self._candidates_phi[atomi] = numpy.array(self._candidates_phi[atomi], dtype=float)
+                self._candidates_bigPhi[atomi] = numpy.array(
+                    self._candidates_bigPhi[atomi], dtype=float
+                )
+                if numpy.count_nonzero(self._candidates_phi[atomi] < 0) > 0:
+                    lower_ind = numpy.where(
+                        self._candidates_phi[atomi]
+                        == self._candidates_phi[atomi][self._candidates_phi[atomi] < 0].max()
+                    )[0][0]
+                    lowerbigPhi = self._candidates_bigPhi[atomi][lower_ind]
+                    lowerphi = self._candidates_phi[atomi][lower_ind]
+                else:  # assign some large negative number
+                    lowerbigPhi = -9999999.0
```

Comment:
```suggestion
                    lowerbigPhi = numpy.NINF
```
and same for positive.

Reply:
I've just modified this. Thanks!